core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSource.java

```diff
@@ -2,25 +2,63 @@
 
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
+import com.kin.ecosystem.common.Callback;
 import com.kin.ecosystem.common.KinCallback;
 import com.kin.ecosystem.common.Observer;
 import com.kin.ecosystem.common.exception.BlockchainException;
 import com.kin.ecosystem.common.exception.ClientException;
 import com.kin.ecosystem.common.model.Balance;
+import com.kin.ecosystem.core.network.ApiException;
+import com.kin.ecosystem.core.network.model.MigrationInfo;
 import com.kin.ecosystem.recovery.KeyStoreProvider;
 import java.math.BigDecimal;
-import java.util.LinkedHashSet;
-import java.util.Set;
 import kin.sdk.migration.MigrationManager;
+import kin.sdk.migration.common.KinSdkVersion;
+import kin.sdk.migration.common.exception.DeleteAccountException;
+import kin.sdk.migration.common.exception.OperationFailedException;
 import kin.sdk.migration.common.interfaces.IKinAccount;
 
 public interface BlockchainSource {
 
+	/**
+	 * Get the migration info from the server.
+	 * @param publicAddress the address associated with the migration information.
+	 * @param callback a callback for this method call.
+	 */
+	void getMigrationInfo(String publicAddress, Callback<MigrationInfo, ApiException> callback);
```

Comment:
The `Callback` should be `KinCallback` and the convert from `ApiException` to `KinEcosystemException` on the `BlockchainSourceImpl`